src/game/Tactical/Squads.cc

```diff
@@ -715,7 +715,7 @@ void LoadSquadInfoFromSavedGameFile(HWFILE const f)
 
 		if (id < 0) continue; // skip empty slot
 
-		if (ubFormatVersion == 0) ubSquadID = i / 6; // use the implicit squadID
+		if (uiSaveGameVersion < 102 || ubFormatVersion == 0) ubSquadID = i / 6; // fixed squads of six, if we are loading a save from old versions
```

Comment:
So I think this should be uiSaveGameVersion < 101 || (uiSaveGameVersion == 101 && ubFormatVersion != SQUAD_INFO_FORMAT_VERSION)